crates/core_arch/src/x86/macros.rs

```diff
@@ -247,6 +247,491 @@ macro_rules! constify_imm4_sae {
+            (8, 2, 8) => $expand!(8, 2, 8),
+            (8, 3, 4) => $expand!(8, 3, 4),
+            (8, 3, 8) => $expand!(8, 3, 8),
+            (9, 0, 4) => $expand!(9, 0, 4),
+            (9, 0, 8) => $expand!(9, 0, 8),
+            (9, 1, 4) => $expand!(9, 1, 4),
+            (9, 1, 8) => $expand!(9, 1, 8),
+            (9, 2, 4) => $expand!(9, 2, 4),
+            (9, 2, 8) => $expand!(9, 2, 8),
+            (9, 3, 4) => $expand!(9, 3, 4),
+            (9, 3, 8) => $expand!(9, 3, 8),
+            (10, 0, 4) => $expand!(10, 0, 4),
+            (10, 0, 8) => $expand!(10, 0, 8),
+            (10, 1, 4) => $expand!(10, 1, 4),
+            (10, 1, 8) => $expand!(10, 1, 8),
+            (10, 2, 4) => $expand!(10, 2, 4),
+            (10, 2, 8) => $expand!(10, 2, 8),
+            (10, 3, 4) => $expand!(10, 3, 4),
+            (10, 3, 8) => $expand!(10, 3, 8),
+            (11, 0, 4) => $expand!(11, 0, 4),
+            (11, 0, 8) => $expand!(11, 0, 8),
+            (11, 1, 4) => $expand!(11, 1, 4),
+            (11, 1, 8) => $expand!(11, 1, 8),
+            (11, 2, 4) => $expand!(11, 2, 4),
+            (11, 2, 8) => $expand!(11, 2, 8),
+            (11, 3, 4) => $expand!(11, 3, 4),
+            (11, 3, 8) => $expand!(11, 3, 8),
+            (12, 0, 4) => $expand!(12, 0, 4),
+            (12, 0, 8) => $expand!(12, 0, 8),
+            (12, 1, 4) => $expand!(12, 1, 4),
+            (12, 1, 8) => $expand!(12, 1, 8),
+            (12, 2, 4) => $expand!(12, 2, 4),
+            (12, 2, 8) => $expand!(12, 2, 8),
+            (12, 3, 4) => $expand!(12, 3, 4),
+            (12, 3, 8) => $expand!(12, 3, 8),
+            (13, 0, 4) => $expand!(13, 0, 4),
+            (13, 0, 8) => $expand!(13, 0, 8),
+            (13, 1, 4) => $expand!(13, 1, 4),
+            (13, 1, 8) => $expand!(13, 1, 8),
+            (13, 2, 4) => $expand!(13, 2, 4),
+            (13, 2, 8) => $expand!(13, 2, 8),
+            (13, 3, 4) => $expand!(13, 3, 4),
+            (13, 3, 8) => $expand!(13, 3, 8),
+            (14, 0, 4) => $expand!(14, 0, 4),
+            (14, 0, 8) => $expand!(14, 0, 8),
+            (14, 1, 4) => $expand!(14, 1, 4),
+            (14, 1, 8) => $expand!(14, 1, 8),
+            (14, 2, 4) => $expand!(14, 2, 4),
+            (14, 2, 8) => $expand!(14, 2, 8),
+            (14, 3, 4) => $expand!(14, 3, 4),
+            (14, 3, 8) => $expand!(14, 3, 8),
+            (15, 0, 4) => $expand!(15, 0, 4),
+            (15, 0, 8) => $expand!(15, 0, 8),
+            (15, 1, 4) => $expand!(15, 1, 4),
+            (15, 1, 8) => $expand!(15, 1, 8),
+            (15, 2, 4) => $expand!(15, 2, 4),
+            (15, 2, 8) => $expand!(15, 2, 8),
+            (15, 3, 4) => $expand!(15, 3, 4),
+            (15, 3, 8) => $expand!(15, 3, 8),
+            (_, _, _) => panic!("Invalid sae value"),
+        }
+    };
+}
+
+// Constifies 8 bits along with an sae option without rounding control.
+// This macro enforces that.
+#[allow(unused)]
+macro_rules! constify_imm8_sae {
```

Comment:
How is this different from just `constify_imm8`?

Reply:
But this macro only has `($imm8:expr, $expand:ident)`, just like `consitfy_imm8`.

Reply:
Ah OK. I think the naming could be clearer, or at least a comment explaining the difference.